### PR TITLE
fix(k8spspprocmount): exemptImages CEL bug

### PR DESCRIPTION
**Pull request by julianKatz**
I recently found (#584) that some K8sNativeValidation implementations of certain templates that iterate over and exempt containers by image had a bug preventing the exemption logic from working.

I've fixed that bug here by mapping from `container` struct to `container.image` string.  I've also added a suite test to verify this. That case fails without the change to the CEL logic.

### CODE REVIEW

File: artifacthub/library/pod-security-policy/proc-mount/1.1.1/samples/psp-proc-mount/example_disallowed_exempt_image.yaml
Comment (JaydipGabani) [inline artifacthub/library/pod-security-policy/proc-mount/1.1.1/samples/psp-proc-mount/example_disallowed_exempt_image.yaml:1]: This should be renamed to `example_allowed_exempt_image.yaml` as we are not expecting it to generate violation. Renaming the file should also fix the CI.

**Review by JaydipGabani — CHANGES_REQUESTED** [review]

File: artifacthub/library/pod-security-policy/proc-mount/1.1.1/samples/psp-proc-mount/example_disallowed_exempt_image.yaml
Comment (julianKatz) [reply artifacthub/library/pod-security-policy/proc-mount/1.1.1/samples/psp-proc-mount/example_disallowed_exempt_image.yaml:1]: Updated this to be an `example_allowed` as you suggested

**Comment by julianKatz** [comment]
Thanks for the review @JaydipGabani!  This is ready for another look.

**Review by JaydipGabani — APPROVED** [review]
lgtm

**Review by apeabody — APPROVED** [review]
LGTM

**Review by maxsmythe — APPROVED** [review]
LGTM

**Comment by Markieta** [comment]
@julianKatz I just pulled this update and it seems to break functionality getting the following violations:

```
failed expression: (has(request.operation) && request.operation == "UPDATE") || size(variables.badContainers) == 0
```

**Pull request merged**